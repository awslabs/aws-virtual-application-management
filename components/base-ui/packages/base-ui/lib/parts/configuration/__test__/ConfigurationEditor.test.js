/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ConfigurationEditor from '../ConfigurationEditor';

describe('ConfigurationEditor', () => {
  it('renders correctly', () =>
    expect(
      renderer.create(<ConfigurationEditor model={{ form: { errors: () => undefined } }} />).toJSON(),
    ).toMatchSnapshot());

  describe('handles click events correctly', () => {
    const mockEvent = { preventDefault: jest.fn(), stopPropagation: jest.fn() };
    const form = { errors: () => [] };
    const onCancel = jest.fn();
    const cancel = jest.fn();
    const next = jest.fn();
    const entries = () => {
      return [['key', 'value']];
    };
    const configurationEditor = shallow(
      <ConfigurationEditor
        onCancel={onCancel}
        model={{
          form,
          hasPrevious: true,
          totalSections: 5,
          currentSectionIndex: 0,
          inputManifestSection: {
            children: [1, 2, 3, 4],
          },
          next,
          cancel,
          definedConfigList: [],
          configuration: { entries },
          review: 'review',
        }}
      />,
    );

    it('handles cancel correctly', async () => {
      configurationEditor.find('Form').simulate('cancel', mockEvent);
      expect(onCancel).toHaveBeenCalledTimes(1);
    });

    it('handles success correctly', async () => {
      configurationEditor.find('Form').simulate('success', mockEvent);
      expect(next).toHaveBeenCalledTimes(1);
    });
  });
});
