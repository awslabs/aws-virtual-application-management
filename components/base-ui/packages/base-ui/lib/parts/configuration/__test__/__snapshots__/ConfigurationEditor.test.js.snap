// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurationEditor renders correctly 1`] = `
<form
  className="ui fluid form"
  onSubmit={[Function]}
>
  <div
    className="ui inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui inverted text loader"
      >
        Processing
      </div>
    </div>
  </div>
  <div
    className="ui padded segment mb3"
  >
    No configuration values are provided
  </div>
  <div
    className="mt0 clearfix"
  >
    <button
      className="ui blue icon right floated right labeled button ml2"
      onClick={[Function]}
      type="submit"
    >
      Next
      <i
        aria-hidden="true"
        className="right arrow icon"
        onClick={[Function]}
      />
    </button>
    <button
      className="ui right floated button"
      onClick={[Function]}
    >
      Clear
    </button>
    <button
      className="ui left floated button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</form>
`;
